File: backend/src/voice_notes/models/schemas/content.py
"""Content schemas for request/response handling."""

from uuid import UUID

from pydantic import BaseModel


class ContentCreate(BaseModel):
    """Schema for creating content."""

    note_id: UUID
    title: str
    content_type: str
    body: str


class ContentUpdate(BaseModel):
    """Schema for updating content."""

    title: str | None = None
    content_type: str | None = None
    body: str | None = None

File: frontend/src/types/content.ts
export interface ContentItem {
  id: string
  title: string
  content_type: string
  body: string
  created_at: string
  note: {
    id: string
    title: string
    transcription: string
  }
}

export interface ContentCreate {
  note_id: string
  title: string
  content_type: string
  body: string
}

export interface ContentGenerateRequest {
  note_ids: string[]
  content_type: string
}

export interface ContentUpdate {
  title?: string
  content_type?: string
  body?: string
}

export interface ContentPageProps {
  content?: ContentItem[]
  isLoading?: boolean
  onCreateClick?: () => void
  onDelete?: (contentId: string) => void
}
